src/dms/frontend/EdFi.DataManagementService.Frontend.AspNetCore/Modules/TokenEndpointModule.cs

```diff
@@ -14,11 +17,28 @@ public void MapEndpoints(IEndpointRouteBuilder endpoints)
         endpoints.MapPost("/oauth/token", GenerateToken);
     }
 
-    internal static async Task GenerateToken(HttpContext httpContext)
+    internal static async Task GenerateToken(HttpContext httpContext, IOptions<AppSettings> appSettings, IOAuthManager oAuthManager, ILogger<TokenEndpointModule> logger, IHttpClientFactory httpClientFactory)
     {
-        var tokenDetails = new TokenResponse("temporary-fake-token", 1800, "bearer");
-        httpContext.Response.StatusCode = StatusCodes.Status200OK;
-        await httpContext.Response.WriteAsSerializedJsonAsync(tokenDetails);
+        // Create client for sending upstream request.
+        logger.LogInformation("Handling OAuth Token request.");
+        var client = httpClientFactory.CreateClient();
+
+        // Extract the Authorization Headers.
+        httpContext.Request.Headers.TryGetValue("Authorization", out var authHeader);
+        try
+        {
+            logger.LogInformation("Requesting access token from OAuth Manager.");
+            var response = await oAuthManager.GetAccessTokenAsync(client, authHeader.ToString(), appSettings.Value.AuthenticationService);
+
+            logger.LogInformation("OAuth Manager access token request successful.");
+            await response.Content.CopyToAsync(httpContext.Response.Body);
+        }
+        catch (OAuthIdentityException ex)
+        {
+            logger.LogError(ex, "Error from OAuthManager service");
+            httpContext.Response.StatusCode = (int?)ex.StatusCode ?? (int)HttpStatusCode.BadGateway;
+            await httpContext.Response.WriteAsync($"Error Getting Access Token Async: {ex.Message}");
+        }
     }
 }
```

Comment:
It looks like we're no longer using `TokenRespose` at the bottom of this file, you can remove it.

Reply:
Removed. Good catch